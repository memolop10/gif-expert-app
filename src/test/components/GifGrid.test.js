import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import {useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");


describe('Test in componente <GifGrid/>', () => {

    const category = 'Genos';

    test('should render correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true 
        })
        
        const wrapper = shallow( <GifGrid category={ category }/> )
        expect( wrapper ).toMatchSnapshot();

    })

    test('should show items when load images useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            title: "Cualquier Cosa",
            url: "https://localhost/goku.jpg"
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false 
        })

        const wrapper = shallow( <GifGrid category={ category }/> )
        expect( wrapper ).toMatchSnapshot();
    })
    
    
});
