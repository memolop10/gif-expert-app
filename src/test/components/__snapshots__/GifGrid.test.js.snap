// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in componente <GifGrid/> should render correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Genos
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test in componente <GifGrid/> should show items when load images useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Genos
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/goku.jpg"
    />
  </div>
</Fragment>
`;
